lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -1220,14 +1220,18 @@ optional<RuleToProcess> SessionState::remove_dynamic_rule(
     const std::string& rule_id, PolicyRule* rule_out,
     SessionStateUpdateCriteria* session_uc) {
   PolicyRule rule;
-  bool removed = dynamic_rules_.remove_rule(rule_id, &rule);
-  if (!removed) {
-    return {};
+  if (!is_5g_session()) {
+    bool removed = dynamic_rules_.remove_rule(rule_id, &rule);
+    if (!removed) {
+      return {};
+    }
+  } else {
+    dynamic_rules_.get_rule(rule_id, &rule);
   }
   if (rule_out) {
     *rule_out = rule;
   }
-  if (session_uc) {
+  if (session_uc && !is_5g_session()) {
```

Comment:
this is becoming very complex. Can we divide this function into 4g and 5g removal?

Reply:
Other than 4G & 5G remove rules use case, This function `SessionState::remove_dynamic_rule` also being called by other scenarios which are common for both 4G & 5G like below:
`apply_session_dynamic_rule_set`
`remove_all_rules_for_termination`
`process_dynamic_rule_installs`
`sync_rules_to_time`
So, every place we have to differentiate 4G/5G which could be more complex than above one. Instead of this I introduced code like above. Please share your suggestions here.

Reply:
What I was saying is keep the signature the same. Just make this function to run 2 other functions to remove the rules if they are 4g or 5g

Bassically, wrap each action in a different function. If you don't want to use functions, you can separate the code into the 4g and the 5g logic.

Note in a year you may be reading this code and not understanding so well. So the idea is to come up with an easy code to read here